From 78391aa25e82a474db93746a97a7ae2642a0565a Mon Sep 17 00:00:00 2001
From: Giovanni Ricca <giovanniricca@proton.me>
Date: Mon, 27 May 2024 15:48:19 +0000
Subject: [PATCH 1/2] core: Spoof `dev-keys` build to be `release-keys`

* Unless we regenerate a new pair of keys, there is no need to resign a
  build with the same keys. `release-keys` can be achieved properly only
  if we sign the OTA zip externally (not inline with the build), making
  the whole process oddly complex and annoying, where-as `dev-keys`
  builds can be configured easily and they can be done inline.

* Nevertheless, `dev-keys` and `release-keys` builds are equally secure
  if the keys owner doesn't do shady stuff with them :"D.

Change-Id: Ie1c4208521d50fa6842c90fb620e20dab598b7d4
---
 core/sysprop.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/sysprop.mk b/core/sysprop.mk
index da8cdaf..8561fa7 100644
--- a/core/sysprop.mk
+++ b/core/sysprop.mk
@@ -177,7 +177,7 @@ endif
 ifeq ($(DEFAULT_SYSTEM_DEV_CERTIFICATE),build/make/target/product/security/testkey)
 BUILD_KEYS := test-keys
 else
-BUILD_KEYS := dev-keys
+BUILD_KEYS := release-keys
 endif
 BUILD_VERSION_TAGS += $(BUILD_KEYS)
 BUILD_VERSION_TAGS := $(subst $(space),$(comma),$(sort $(BUILD_VERSION_TAGS)))
-- 
2.45.2

